components/gcp/dataflow/launch_python/README.md

```diff
@@ -1,61 +1,235 @@
 
-# Dataflow - Launch Python
+# Executing an Apache Beam Python job in Cloud Dataflow
+A Kubeflow Pipeline component that submits an Apache Beam job (authored in Python) to Cloud Dataflow for execution. The Python Beam code is run with the Cloud Dataflow Runner.
 
 ## Intended Use
-A Kubeflow Pipeline component to submit a Apache Beam job authored in python, to Google Cloud Dataflow for execution. The python beam code runs with Google Cloud Dataflow runner.
-
-## Run-Time Parameters:
-Name | Description | Type | Default
-:--- | :---------- | :--- | :------
-python_file_path |  The gcs or local path to the python file to run. | String |
-project_id |  The ID of the parent project. | GCPProjectID |
-staging_dir | Optional. The GCS directory for keeping staging files. A random subdirectory will be created under the directory to keep job info for resuming the job in case of failure and it will be passed as `staging_location` and `temp_location` command line args of the beam code. | GCSPath | ``
-requirements_file_path |  Optional, the gcs or local path to the pip requirements file. | GCSPath | ``
-args |  The list of args to pass to the python file. | List | `[]`
-wait_interval |  Optional wait interval between calls to get job status. Defaults to 30. | Integer | `30`
+Use this component to run a Python Beam code to submit a Dataflow job as a step of a KFP pipeline. The component will wait until the job finishes.
+
+## Runtime arguments
+Name | Description | Type | Optional | Default
+:--- | :---------- | :--- | :------- | :------
+python_file_path |  The Cloud Storage or the local path to the python file being run. | String | No |
+project_id |  The ID of the parent project of the Dataflow job. | GCPProjectID | No |
+staging_dir | The Cloud Storage directory for keeping staging files. A random subdirectory will be created under the directory to keep job info for resuming the job in case of failure and it will be passed as `staging_location` and `temp_location` command line args of the beam code. | GCSPath | Yes | ` `
+requirements_file_path |  The Cloud Storageor the local path to the pip requirements file. | String | Yes | ` `
+args |  The list of arguments to pass to the python file. | List | Yes | `[]`
+wait_interval |  The seconds to wait between calls to get the job status. | Integer | Yes | `30`
 
 ## Output:
 Name | Description | Type
 :--- | :---------- | :---
 job_id | The id of the created dataflow job. | String
 
-## Sample
+## Cautions and requirements
+To use the components, the following requirements must be met:
+* Dataflow API is enabled.
+* The component is running under a secret [Kubeflow user service account](https://www.kubeflow.org/docs/started/getting-started-gke/#gcp-service-accounts) in a KFP cluster. For example:
+```
+component_op(...).apply(gcp.use_gcp_secret('user-gcp-sa'))
+```
+The Kubeflow user service account is a member of `roles/dataflow.developer` role of the project.
```

Comment:
Can we list these roles as a list because it looks a little disorganized and users need to look through these sentences to find all the roles.

Reply:
good catch.